stack/core/src/main/java/org/apache/usergrid/corepersistence/CpSetup.java

```diff
@@ -61,151 +59,106 @@
+            logger.warn( "Application {}/{} already exists", DEFAULT_ORGANIZATION, DEFAULT_APPLICATION );
+        }
+        catch ( OrganizationAlreadyExistsException oaee ) {
+            logger.warn( "Organization {} already exists", DEFAULT_ORGANIZATION );
         }
 
-        Injector injector = CpSetup.getInjector();
-        MigrationManager m = injector.getInstance( MigrationManager.class );
         try {
-            m.migrate();
-        } catch (MigrationException ex) {
-            throw new RuntimeException("Error migrating Core Persistence", ex);
+            emf.initializeApplication( DEFAULT_ORGANIZATION, emf.getManagementAppId(), MANAGEMENT_APPLICATION, null );
+        }
+        catch ( ApplicationAlreadyExistsException ex ) {
+            logger.warn( "Application {}/{} already exists", DEFAULT_ORGANIZATION, MANAGEMENT_APPLICATION );
+        }
+        catch ( OrganizationAlreadyExistsException oaee ) {
+            logger.warn( "Organization {} already exists", DEFAULT_ORGANIZATION );
         }
-
-        setupSystemKeyspace();
-
-        setupStaticKeyspace();
-
-        //force the EMF creation of indexes before creating the default applications
-        emf.refreshIndex();
-
-        createDefaultApplications();
     }
 
 
-    public void createDefaultApplications() throws Exception {
-
-        logger.info("Setting up default applications");
-
+    /**
+     * Perform migration of the 2.0 code
+     */
+    private void migrate() {
+        MigrationManager m = injector.getInstance( MigrationManager.class );
         try {
-            emf.initializeApplication( DEFAULT_ORGANIZATION,
-                    emf.getDefaultAppId(), DEFAULT_APPLICATION, null );
-        } catch (ApplicationAlreadyExistsException ex) {
-            logger.warn("Application {}/{} already exists", 
-                    DEFAULT_ORGANIZATION, DEFAULT_APPLICATION);
+            m.migrate();
         }
-
-        try {
-            emf.initializeApplication( DEFAULT_ORGANIZATION,
-                    emf.getManagementAppId(), MANAGEMENT_APPLICATION, null );
-        } catch (ApplicationAlreadyExistsException ex) {
-            logger.warn("Application {}/{} already exists", 
-                    DEFAULT_ORGANIZATION, MANAGEMENT_APPLICATION);
+        catch ( MigrationException ex ) {
+            throw new RuntimeException( "Error migrating Core Persistence", ex );
         }
     }
 
 
-
-    
     @Override
     public void setupSystemKeyspace() throws Exception {
 
         logger.info( "Initialize system keyspace" );
 
-        cass.createColumnFamily( SYSTEM_KEYSPACE, createColumnFamilyDefinition( 
-                SYSTEM_KEYSPACE, APPLICATIONS_CF, ComparatorType.BYTESTYPE ) );
+        migrate();
 
-        cass.createColumnFamily( SYSTEM_KEYSPACE, createColumnFamilyDefinition( 
-                SYSTEM_KEYSPACE, PROPERTIES_CF, ComparatorType.BYTESTYPE ) );
+        cass.createColumnFamily( SYSTEM_KEYSPACE,
```

Comment:
Is there a way to check that we've already created these keyspaces? That way it won't show the exception when we try to recreate already existing keyspaces?